test/testcmdlineparser.cpp

```diff
@@ -421,6 +422,19 @@ class TestCmdlineParser : public TestFixture {
         (void)logger->str(); //ASSERT_EQUALS("The Product\n", logger->str()); // TODO: include version?
     }
 
+    void versionSwitchWithCfg() {
+        REDIRECT;
+        ScopedFile file(Path::join(Path::getPathFromFilename(Path::getCurrentExecutablePath("")), "cppcheck.cfg"),
+                        "{\n"
+                        "\"productName\": \"Cppcheck Premium 1.2.3.4\""
+                        "}\n");
+        const char * const argv[] = {"cppcheck", "--premium=misra-c++-2008", "--version", "file.cpp"};
+        // --premium=misra-c++-2008 should be recognized from cfg
+        ASSERT_EQUALS(CmdLineParser::Result::Exit, parser->parseFromArgs(4, argv));
```

Comment:
somehow you need to enforce that the cppcheck.cfg is not in the current working path.